OpenRA.Mods.Common/Traits/Render/WithDeathAnimation.cs

```diff
@@ -71,10 +74,23 @@ public WithDeathAnimation(Actor self, WithDeathAnimationInfo info)
 
 		public void Killed(Actor self, AttackInfo e)
 		{
-			// Killed by some non-standard means. This includes being crushed
-			// by a vehicle (Actors with Crushable trait will spawn CrushedSequence instead).
+			// Actors with Crushable trait will spawn CrushedSequence.
+			var crushable = self.TraitOrDefault<Crushable>();
+			if (crushable != null && crushable.Crushed)
```

Comment:
The new boolean seems redundant. Why not just implement `ICrushable.OnCrush` here? Also saves the trait lookup.